File: .guides/tests/py-2.js

var fileName = window.location.origin + '/public/content/py/' + window.testEnv.cmd + '.py';

$.get(fileName)
.success(function(data) {

  if(data == 404) {
    console.log(exception);
    codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.INVALID, fileName + ' not found'); 
  }

  try {        
    var module = Sk.importMainWithBody("", false, data);

    if(indexOfCommand('createEmptyMaze') > -1 && indexOfCommand('addRandomMonsters') > -1 && command('addRandomMonsters')[1] == 3) {
      codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.SUCCESS, 'Well done!');
    }
    else {
      codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
    }
  }
  catch(exception) {
    console.log('skulpt exception');
    console.log(exception);
    codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.INVALID, exception.toString()); 
  }
})
.fail(function(jqxhr, settings, exception) {
  console.log('jqxhr fail');
  console.log(exception);
  codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.INVALID, exception.message); 
});

File: .guides/tests/py-2.py

maze = False
player = False
monsterCount = False

def createEmptyMaze():
  global maze
  maze = True

def addRandomMonsters(count):
  global monsterCount
  monsterCount = count

def addPlayer():
  global player
  player = True

try:
  execfile('/home/codio/workspace/public/content/py/py-2.py')
  
  if maze == True and monsterCount == 3:
    print 'well done'
    exit(0)
except (IOError, SyntaxError) as e:
  pass
  
print 'Not quite right, try again!'
exit(1)
